File: nim/darts/Iteration1/darts.nim
import std/math
  
proc distanceFromCentre(x, y: float): float =
  sqrt(pow(x, 2.0) + pow(y, 2.0))
  
proc score*(x, y: float): int =
  let dist = distanceFromCentre(x, y);

  if (dist >= 0.0 and dist <= 1.0):
    return 10
  elif (dist >= 1.0 and dist <= 5.0):
    return 5
  elif (dist >= 5.0 and dist <= 10.0):
    return 1
  else:
    return 0

File: zig/darts/Iteration1/darts.zig
const std = @import("std");
const math = std.math;

const centre:       f32 = 0;
const innerRadius:  f32 = 1;
const middleRadius: f32 = 5;
const outerRadius:  f32 = 10;

pub const Coordinate = struct {
    x_coord: f32,
    y_coord: f32,

    pub fn init(x_coord: f32, y_coord: f32) Coordinate {
        return Coordinate {
            .x_coord = x_coord,
            .y_coord = y_coord,
        };
    }

    fn distanceFromCentre(coord: Coordinate) f32 {
        return math.sqrt(math.pow(f32, coord.x_coord, 2) + math.pow(f32, coord.y_coord, 2));
    }
    
    pub fn score(self: Coordinate) usize {
        const dist = self.distanceFromCentre();

        if (dist >= centre and dist <= innerRadius) return 10;
        if (dist >= innerRadius and dist <= middleRadius) return 5;
        if (dist >= middleRadius and dist <= outerRadius) return 1;
    
        return 0;
    }
};
